packages/client/src/components/app/forms/RelationshipField.svelte

```diff
@@ -108,8 +108,16 @@
     }
   }
 
+  $: forceFetchRows(filter)
   $: debouncedFetchRows(searchTerm, primaryDisplay, defaultValue)
 
+  const forceFetchRows = async () => {
+    // if the filter has changed, then we need to reset the options, clear the selection, and re-fetch
+    optionsObj = {}
```

Comment:
This would clean the `optionsObj` cache. This was added to fix initial value fetches, so we should never do this. The filters should be happening on memory as it's currently doing

Reply:
The filter object can change though, such as in this example. If the `optionsObj` isn't cleared then it will add an additional page of filtered options rather than override the existing options with the filtered options